rocketpy/rocket/rocket.py

```diff
@@ -525,18 +536,50 @@ def evaluate_center_of_pressure(self):
         # Calculate total lift coefficient derivative and center of pressure
         if len(self.aerodynamic_surfaces) > 0:
             for aero_surface, position in self.aerodynamic_surfaces:
+                if isinstance(aero_surface, GenericSurface):
+                    continue
                 # ref_factor corrects lift for different reference areas
                 ref_factor = (aero_surface.rocket_radius / self.radius) ** 2
                 self.total_lift_coeff_der += ref_factor * aero_surface.clalpha
                 self.cp_position += (
                     ref_factor
                     * aero_surface.clalpha
-                    * (position - self._csys * aero_surface.cpz)
+                    * (position.z - self._csys * aero_surface.cpz)
                 )
-            self.cp_position /= self.total_lift_coeff_der
-
+            # Avoid errors when only generic surfaces are added
+            if self.total_lift_coeff_der.get_value(0) != 0:
+                self.cp_position /= self.total_lift_coeff_der
         return self.cp_position
 
+    def evaluate_surfaces_cp_to_cdm(self):
+        """Calculates the relative position of each aerodynamic surface center
+        of pressure to the rocket's center of dry mass in Body Axes Coordinate
+        System.
+
+        Returns
+        -------
+        self.surfaces_cp_to_cdm : dict
+            Dictionary mapping the relative position of each aerodynamic
+            surface center of pressure to the rocket's center of mass.
+        """
+        for surface, position in self.aerodynamic_surfaces:
+            self.evaluate_single_surface_cp_to_cdm(surface, position)
+        return self.surfaces_cp_to_cdm
+
+    def evaluate_single_surface_cp_to_cdm(self, surface, position):
```

Comment:
I'm under the impression this second method should be private.

The reason: why should we provide the public API with 2 different methods that does basically the same thing? It is similar to the discussion we had in the past with `add_surface(s)` methods

```suggestion
    def __evaluate_single_surface_cp_to_cdm(self, surface, position):
```